pkg/controller/controller.go

```diff
@@ -605,3 +607,41 @@ func unsetLoggerCtxValue(ctx context.Context, key string) context.Context {
 
 	return ctx
 }
+
+func reportLastReconciled(o interface{}) {
+	var kind string
+	{
+		obj, ok := o.(pkgruntime.Object)
+		if !ok {
+			return
+		}
+
+		gvks, _, err := scheme.Scheme.ObjectKinds(obj)
```

Comment:
Actually, If I could use `meta.TypeAccessor` (https://godoc.org/k8s.io/apimachinery/pkg/api/meta#TypeAccessor), it would be much easier to fetch `kind` information, but actually all `TypeMeta` is empty in default so I need to use this `ObjectKinds` instead.

Reply:
For my interest do you know if the controller runtime client sets this type information?

It's a large change but I think in sig-operator we wanted to move away from the generated client to the controller client. This might be another reason for that change.

Reply:
operatorkit collect another timestamp metric and it is using controller runtime, and I see they have no problem with type information. https://github.com/giantswarm/operatorkit/blob/dc6f5aa4fb98c5cc3b42bfada11793e41144cbdd/pkg/controller/collector/timestamp.go#L60-L87

So you're right at this point; this is another reason we need to move to controller runtime client.

Reply:
Cool I'll find the SIG issue and post there. I'll also create an issue in the bat board.